express/features/support/steps/shared-functions.ts

```diff
@@ -23,6 +23,18 @@ export async function getButtonLink(page: Page, linkText: string): Promise<Eleme
     return getSingleElement(links, linkText);
 }
 
+export async function getButtonLinkWithClass(page: Page, linkText: string): Promise<ElementHandle> {
+    const links = await page.$x(`//button[contains(text(), "${linkText}")][@class="gem-c-show-password__toggle"]`);
+    return getSingleElement(links, linkText);
+}
```

Comment:
Make `class` a param in the `getButtonLink` method with the default value of `govuk-button` instead of duplicating the method. The logic is the same for both, we just need to be able to override the default class.

Reply:
We can pass the default params and override on the call. But expression that searches the element is different

- //button[contains(text(), "${linkText}")][@class="${buttonClass}"]
- //button[contains(text(), "${linkText}")][@class="${buttonClass}"][@aria-controls="${fieldName}"]

I guess we need different functions unless we want to put some logic inside the method. Something like on availability of overridden param - change the expression and add @aria-controls. Which is not that neat I suppose.